### PR TITLE
Add time range parameters to labels API

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixes: #1811

* [x] Add changelog
* [x] Update unit tests and E2E tests.

### CODE REVIEW

**Comment by yeya24** [comment]
This pr is ready for review!

File: pkg/api/query/v1.go [hunk 1]
```diff
@@ -394,12 +388,27 @@ func (qapi *QueryAPI) labelValues(r *http.Request) (interface{}, []error, *api.A
...
 
+	start, err := parseTimeParam(r, "start", minTime)
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+	}
+	end, err := parseTimeParam(r, "end", maxTime)
+	if err != nil {
+		return nil, nil, &api.ApiError{Typ: api.ErrorBadData, Err: err}
+	}
+	// Prometheus doesn't check this, do we need to?
```
Comment (bwplotka) [inline pkg/api/query/v1.go:399]: Sounds useful, no? (: 

Please ask comments on PR itself not in comment (: So we can still merge PR if code is ok.

File: pkg/store/storepb/rpc.proto [hunk 2]
```diff
@@ -144,6 +144,10 @@ message LabelNamesRequest {
 
   // TODO(bwplotka): Move Thanos components to use strategy instead. Including QueryAPI.
   PartialResponseStrategy partial_response_strategy = 2;
+
+  int64 start = 3;
```
Comment (bwplotka) [inline pkg/store/storepb/rpc.proto:148]: :call_me_hand:

File: pkg/api/query/v1.go (hunk 1 above)
Comment (GiedriusS) [reply pkg/api/query/v1.go:399]: I think we should and it's probably worth contributing such a check upstream IMHO

**Comment by yeya24** [comment]
I have addressed the comment. Please take a look again. :smile:

**Pull request merged**